deployment/src/main/java/io/quarkiverse/jasperreports/deployment/JasperReportsProcessor.java

```diff
@@ -570,20 +570,8 @@ void compileReports(ReportBuildTimeConfig config, List<ReportFileBuildItem> repo
                     String outputFile = outputFilePath.toString();
 
                     try {
-                        boolean shouldCompile = true;
-
-                        // Check if output file exists
-                        if (Files.exists(outputFilePath)) {
-                            // Compare last modified times
-                            FileTime inputFileLastModifiedTime = Files.getLastModifiedTime(item.getPath());
-                            FileTime outputFileLastModifiedTime = Files.getLastModifiedTime(outputFilePath);
-
-                            // Skip compilation if output file is newer or the same as the input file
-                            if (outputFileLastModifiedTime.compareTo(inputFileLastModifiedTime) >= 0) {
-                                shouldCompile = false;
-                                Log.debugf("Skipping compilation, output file %s is up to date.", outputFile);
-                            }
-                        }
+                        // Use the helper method to check if the output file needs updating
+                        boolean shouldCompile = shouldUpdateFile(item.getPath(), outputFilePath);
```

Comment:
I'd probably put the method call directly in the `if` statement since the local variable isn't used any where else, but other than that, LGTM.

Reply:
Excellent feedback. Done!